Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py

```diff
@@ -499,12 +499,75 @@ def _run_single_test(self, driver, test_input, stop_when_done):
 
 class TestShard(object):
     """A test shard is a named list of TestInputs."""
+    DEFAULT_WEIGHT = 256
+    WEIGHT_ESTIMATES = {
+        'animations': 438,
```

Comment:
Do we want the (presumably exact) current number, or do we want to approximate it and round it (to multiples of 256, to powers of 2, etc.)?

Reply:
Prefacing this comment by pointing out that this change was originally just meant as an experiment (which is why it's still marked as a "draft"). That being said, the "experiment" was surprisingly successful (10-15% runtime improvement, no side effects), so it seems likely that a version of this change will find it's way up for review in short order.

These numbers are the average time, in milliseconds, that each individual test takes to complete. The average for the whole suite is around 200. The thinking here is that we can get a pretty accurate estimate of shard time as long as our guess for how long tests in a given shard will take isn't orders of magnitude off. I actually calculated average tests time for every shard, but if a child's average test time was within 25% of its parent's, I deleted the child. I think did a bit of manual curation, deleting some measurements if there weren't very many tests in a shard.

Reply:
I think I'm still in favour of rounding this further than the current raw figures; yes, it'll make it less accurate, but I suspect we have more precision here than the noise really allows us to justify.